src/openrct2/core/String.cpp

```diff
@@ -343,65 +338,60 @@ namespace String
         return buffer;
     }
 
-    utf8* Format(const utf8* format, ...)
+    std::string Format(const utf8* format, ...)
     {
         va_list args;
         va_start(args, format);
-        utf8* result = Format_VA(format, args);
+        std::string result = Format_VA(format, args);
         va_end(args);
         return result;
     }
 
-    utf8* Format_VA(const utf8* format, va_list args)
+    std::string Format_VA(const utf8* format, va_list args)
     {
         va_list args1, args2;
         va_copy(args1, args);
         va_copy(args2, args);
 
         // Try to format to a initial buffer, enlarge if not big enough
         size_t bufferSize = 4096;
-        utf8* buffer = Memory::Allocate<utf8>(bufferSize);
+        std::string buffer;
+        buffer.resize(bufferSize - 1);
 
-        // Start with initial buffer
-        int32_t len = vsnprintf(buffer, bufferSize, format, args);
+        int32_t len = vsnprintf(&buffer[0], bufferSize, format, args);
```

Comment:
Drop the hardcoded buffer length. `vsnprintf` returns the length required if you pass it no length. Simply use `len` to resize the buffer.